### PR TITLE
nim: update to 1.4.0

### PR DESCRIPTION
**Pull request by IcedQuinn**
Current Nim is about a year out of date.

### CODE REVIEW

File: srcpkgs/nim/template [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-revision=2
+version=1.4.0
+revision=1
 _cversion=0.20.0
-_nimbleversion=0.11.4
+_nimbleversion=0.12.0
 create_wrksrc=yes
 build_wrksrc="Nim-$version"
-hostmakedepends="ed"
+hostmakedepends="ed git"
```
Comment (fosslinux) [inline srcpkgs/nim/template:9]: What is this needed for

Comment (IcedQuinn) [reply srcpkgs/nim/template:9]: the bootstrap script downloads the "fusion" package ( https://github.com/nim-lang/fusion )

technically nimble also soft-requires git, being a package manager that almost entirely works through github in practice.

**Comment by rc-05** [comment]
There's already a PR that I've done about the nim package (#25676) so there's no need to duplicate stuff 🙂

**Comment by IcedQuinn** [comment]
> There's already a PR that I've done about the nim package (#25676) so there's no need to duplicate stuff 

Doesn't seem to have any movement on it for some months either :cold_sweat: 

> Maybe I can get it to also make use of clang as an alternative (via build option) since GCC seems outdated (it's latest version is 10.2)

Void does't use GCC?

**Comment by ericonr** [comment]
Does Nim require something from GCC 10.2? 9.3 isn't that old.

**Comment by IcedQuinn** [comment]
> Does Nim require something from GCC 10.2? 9.3 isn't that old.

not that i know of.

**Comment by fosslinux** [comment]
Don't use clang unless it is explicitly needed to build..

**Comment by IcedQuinn** [comment]
is there a particular blocker that should be fixed?

**Comment by rc-05** [comment]
> Does Nim require something from GCC 10.2? 9.3 isn't that old.

Nim uses GCC primarly but in reality you can use any fairly modern C compiler as the backend fortunately.

**Comment by rc-05** [comment]
> Don't use clang unless it is explicitly needed to build..

No problem, will leave as it is :+1:

**Comment by IcedQuinn** [comment]
:question:

**Comment by ahesford** [comment]
You don't need `git` if you fetch a specific `fusion` commit, as is done with `nimble`. This will be superseded by #26868, which should be merged shortly.

**Pull request closed without merging**